guidelines/monai-application-package.md

```diff
@@ -1,498 +1,650 @@
-> These items are suggested, but not part of any specification yet.
+- The Application Manifest SHOULD define the output path relative to the working directory used by the Application (`/etc/holoscan/app.json#output.path`).
 
-- Application Manifest MUST define the command used to run the Application (`/etc/monai/app.json#command`).
+  - The output path SHOULD be a relative file-system path to a directory or folder.
 
-  - When not provided the MAP will be considered invalid and not runnable by MONAI Deploy Application Runner and Server.
+    - When the value is a relative file-system path (the first character is not `/`), it is relative to the application's working directory.
 
-- Application Manifest SHOULD define the version of the manifest file schema (`/etc/monai/app.json#api-version`).
+    - When the value is an absolute file-system path (the first character is `/`), the file-system path is used as-is.
 
-  - Manifest schema version SHALL be provided as a [semantic version](https://semver.org/) string.
+  - When not provided, the default value `output/` SHALL be assumed.
 
-  - When not provided the default value `0.0.0` SHALL be assumed.
+- The Application Manifest SHOULD define the output data format produced by the Application (`/etc/holoscan/app.json#output.format`).
 
-- Application Manifest SHOULD define the version of the MONAI Deploy SDK used to create the Application (`/etc/monai/app.json#sdk-version`).
+  - Possible values include, but are not limited to, `none`, `screen`, `file`, `network`.
 
-  - SDK version SHALL be provided as a [semantic version](https://semver.org/) string.
+- The Application Manifest SHOULD configure a check to determine whether or not the application is "ready."
 
-  - When not provided the default value `0.0.0` SHALL be assumed.
+  - The Application Manifest SHALL define the probe type to be performed (`/etc/holoscan/app.json#readiness.type`).
 
-- Application Manifest SHOULD define the version of the Application itself (`/etc/monai/app.json#version`).
+    - Possible values include `tcp`, `grpc`, `http-get`, and `command`.
 
-  - Application version SHALL be provided as a [semantic version](https://semver.org/) string.
+  - The Application Manifest SHALL define the probe commands to execute when the type is `command` (`/etc/holoscan/app.json#readiness.command`).
 
-  - When not provided the default value `0.0.0` SHALL be assumed.
+    - The data structure is expected to be an array of strings.
 
-- Application Manifest SHOULD define the Application's working directory (`/etc/monai/app.json#working-directory`).
+  - The Application Manifest SHALL define the port to perform the readiness probe when the type is `grpc`, `tcp`, or `http-get`. (`/etc/holoscan/app.json#readiness.port`)
 
-  - The Application will execute with its current directory set to this value.
+    - The value provided must be a valid port number ranging from 1 through 65535.
```

Comment:
Maybe the range should be 1024 - 65535 to avoid root-only privileged ports.

Reply:
I think we can make a note to let users know about root-only privileged ports instead of limiting them from using them.  E.g., If someone wants a long-running service that hosts some web pages on port 80, that's fine as long as they documented it in their user guide.

Reply:
Well, with the port 80 example, it might be fine for an application to hijack port 80 for its exclusive use on a embedded/dedicated device, however, it is NOT on shared host system, so I would think the example supports the argument.